### PR TITLE
[CodeCamp #2] Add jax support to MeanIoU

### PR DESCRIPTION
**Pull request by xuan07472**
## Motivation

Add jax support to MeanIoU

## Modification

Calculated using jax.numpy.ndarray for MeanIoU support, supplemented and unit tested

### CODE REVIEW

File: tests/test_metrics/test_mean_iou.py [hunk 1]
```diff
@@ -5,12 +5,14 @@
 import numpy as np
 import pytest
 from distutils.version import LooseVersion
+from jax import random
```
Comment (ice-tong) [inline tests/test_metrics/test_mean_iou.py:8]: This should be removed~
This import statement could be error if the JAX is not installed.

**Comment by ice-tong** [comment]
Please update the docstring of MeanIoU~
https://github.com/open-mmlab/mmeval/blob/25a63aa4adbcf5e9ae074bbe09e3b022f18ad96d/mmeval/metrics/mean_iou.py#L32-L34

File: mmeval/metrics/mean_iou.py [hunk 2]
```diff
@@ -29,10 +33,10 @@ class MeanIoU(BaseMetric):
...
     accuracy, mean dice, mean precision, mean recall and mean f-score.
 
-    This metric supports 5 kinds of inputs, i.e. ``numpy.ndarray``,
-    ``torch.Tensor``, ``oneflow.Tensor``, ``tensorflow.Tensor`` and
-    ``paddle.Tensor``, and the implementation for the calculation depends
-    on the inputs type.
+    This metric supports 6 kinds of inputs, i.e. ``numpy.ndarray``,
+    ``torch.Tensor``, ``oneflow.Tensor``, ``tensorflow.Tensor``,
+    ``paddle.Tensor``and``jax.DeviceArray``, and the implementation for
+    the calculation dependson the inputs type.
```
Comment (ice-tong) [inline mmeval/metrics/mean_iou.py:39]:
```suggestion
    ``paddle.Tensor``and``jax.Array``, and the implementation for
    the calculation dependson the inputs type.
```

File: tests/test_metrics/test_mean_iou.py (hunk 1 above)
Comment (xuan07472) [reply tests/test_metrics/test_mean_iou.py]: 感谢解答，已修改，赞！

File: mmeval/metrics/mean_iou.py [hunk 3]
```diff
@@ -210,6 +214,30 @@ def compute_confusion_matrix(  # type: ignore
...
+            numpy.ndarray: The computed confusion matrix.
+        """
+        mask = (label != self.ignore_index)
+        prediction, label = prediction[mask], label[mask]
+        confusion_matrix_1d = jnp.bincount(
+            num_classes * label + prediction, minlength=num_classes**2)
+        confusion_matrix = confusion_matrix_1d.reshape(num_classes,
+                                                       num_classes)
+        confusion_matrix = np.asarray(confusion_matrix)
+        return confusion_matrix
```
Comment (ice-tong) [inline mmeval/metrics/mean_iou.py:239]:
```suggestion
        return np.asarray(confusion_matrix)
```

**Comment by ice-tong** [comment]
LGTM, and please @zhouzaida have a look~

File: mmeval/metrics/mean_iou.py [hunk 4]
```diff
@@ -29,10 +33,10 @@ class MeanIoU(BaseMetric):
...
     accuracy, mean dice, mean precision, mean recall and mean f-score.
 
-    This metric supports 5 kinds of inputs, i.e. ``numpy.ndarray``,
-    ``torch.Tensor``, ``oneflow.Tensor``, ``tensorflow.Tensor`` and
-    ``paddle.Tensor``, and the implementation for the calculation depends
-    on the inputs type.
+    This metric supports 6 kinds of inputs, i.e. ``numpy.ndarray``,
+    ``torch.Tensor``, ``oneflow.Tensor``, ``tensorflow.Tensor``,
+    ``paddle.Tensor``and``jax.Array``, and the implementation for
+    the calculation dependson the inputs type.
```
Comment (zhouzaida) [inline mmeval/metrics/mean_iou.py:39]:
```suggestion
    the calculation depends on the inputs type.
```

**Review by zhouzaida — APPROVED** [review]

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**